.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,51 @@
+name: Cargo build and test
+on: [pull_request, workflow_dispatch]
+jobs:
+  test:
+    strategy:
+      matrix:
+        target:
+          - i686-pc-windows-gnu
+          - x86_64-pc-windows-gnu
+          - i686-pc-windows-msvc
+          - x86_64-pc-windows-msvc
+        rust: [stable]
+        include:
+          # Attach the platform data for MinGW to the -gnu targets
+          - target: i686-pc-windows-gnu
+            platform: x86
+          - target: x86_64-pc-windows-gnu
+            platform: x64
+          # Test more Rust channels on x86_64 msvc
+          - target: x86_64-pc-windows-msvc
+            rust: beta
+          - target: x86_64-pc-windows-msvc
+            rust: nightly
+          - target: x86_64-pc-windows-msvc
+            rust: 1.58.0
+
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up MinGW
+        if: matrix.target == 'i686-pc-windows-gnu' || matrix.target == 'x86_64-pc-windows-gnu'
+        uses: egor-tensin/setup-mingw@v2
+        with:
+          platform: ${{ matrix.platform }}
```

Comment:
I noticed that the CI works without this parameter also. so for some reason `i686-pc-windows-gnu` builds when this is unset. The documentation for this action says that the default is x64, but I guess it installs both toolchains by default then. I guess it does not hurt to specify it more explicitly? Except a bit more verbose CI config... I'm a bit afraid we test the wrong thing without it :thinking: